### DISCUSSION TITLE
About FutureWarnings: .* suffix can only be used with `==` or `!=` operators

### DISCUSSION THREAD

**Discussion by pawamoy** [pdm-project/pdm#2211] [General]
I'm getting one or two warnings like this per installation.

> /home/pawamoy/.local/pipx/venvs/pdm/lib/python3.11/site-packages/unearth/evaluator.py:177: FutureWarning: .* suffix can only be used with `==` or `!=` operators

The issue is that:

1. it's a user warning, but I'm not responsible for it, and I have no control over the library who used such specifiers
2. even if I had control, the warning does not tell me which library is using such specifiers. I cannot even report it to its maintainers.
3. it makes me worry that `pdm install` will start failing at some point because this becomes unsupported?

I think the warning should point at the offensive library, so that we can fix the specifiers there. Also, what's the plan? Will this actually start failing at some point?

(opening here for visibility)

**Comment by pawamoy** [comment]
See https://github.com/pdm-project/pdm/issues/2284#issuecomment-1740697897 for a possible solution. In my case, the warnings came from semver 3.0.0 dev versions which specify requires-python `>=3.6.*` or `>=3.7.*`. I wonder if the warning is useful since these versions will never be patched to modify their requires-python. The warning is simply not actionable in this case. It is only actionable when this is for the latest version of a package. But that is hard to guess. Not sure what's the best thing to do here :shrug:

**Comment by jasonwbarnett** [comment]
I stumbled upon this discussion when trying to hunt down an issue when trying to generate a pex lockfile. I think point number 2 that @pawamoy brought up is huge. How can I determine the issue when I don't even know what the offending library is?
